File: Shank/benchMark/Fibonacci/Fibonacci.py
# Setting INNER_LOOP_LEN to 44 causes the inner loop to calculate the 46th
# Fibonacci number: 1836311903
# See: https://planetmath.org/listoffibonaccinumbers
# The 46th Fibonacci number was chosen as the target because it is the largest
# Fibonacci number that is smaller than INT_MAX (2147483647).

INNER_LOOP_LEN = 44
OUTER_LOOP_LEN = 9000000

def fibonacci(m, n):
    curr = 1
    for i in range(m):
        prev1 = 1
        prev2 = 1
        curr = 1
        for j in range(n):
            curr = prev1 + prev2
            prev2 = prev1
            prev1 = curr
    return curr


print(fibonacci(OUTER_LOOP_LEN, INNER_LOOP_LEN))

File: Shank/benchMark/Fibonacci/Fibonacci.c
#include <stdio.h>

#define INNER_LOOP_LEN 88
#define OUTER_LOOP_LEN 100000000

int main() {
  unsigned long long curr = 1;
  for (int j = 0; j < OUTER_LOOP_LEN; j++) {
    unsigned long long prev1 = 1;
    unsigned long long prev2 = 1;
    curr = 1;
    // This results in the 90th fibonacci number: 2880067194370816120
    // See: https://planetmath.org/listoffibonaccinumbers
    for (int i = 0; i < INNER_LOOP_LEN; i++) {
      curr = prev1 + prev2;
      prev2 = prev1;
      prev1 = curr;
    }
  }
  printf("%llu", curr);
  return 0;
}
